### ISSUE TITLE
the stange D loss training in pix2pix

### ISSUE THREAD

**Issue by 13462877152** [junyanz/pytorch-CycleGAN-and-pix2pix#1129]
Dear professor, Thank you for your excellent papers and projects. There are two ways to train the pix2pix as follows: the first is in readme.md  and the second is in train_pix2pix.sh.  The results and the loss  are quite different. , especially the first one ,the D real and D false are almost zero during in the training . How to choose the two methods? 
1、python train.py --dataroot ./datasets/facades --name facades_pix2pix --model pix2pix --direction BtoA
2、python train.py --dataroot ./datasets/facades --name facades_pix2pix --model pix2pix --netG unet_256 --direction BtoA --lambda_L1 100 --dataset_mode aligned --norm batch --pool_size 0

**Comment by junyanz** [comment]
Not sure what happened. These two training command lines should create the same training instance. The arguments used in the command line are the same as default arguments. The results might be different across different runs in your application. You can double-check it by looking at the printed arguments on the console.

**Comment by 13462877152** [comment]
Thank you very much for your reply.  I am also puzzled about the tanh() in the last layer of the generator. If  the input need to be in [0,1] , is it also necessary to use tanh() in the last  layer.

**Comment by junyanz** [comment]
Tanh() is not affected by your input range. If your output is [0, 1], you can either pre-process your input to [-1, 1], or change the TanH to Sigmoid.

**Comment by 13462877152** [comment]
Thank you very much for your advice, I benefited a lot from the excellent project. It's really great.
